devel/balena-cli/Portfile

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        balena-io balena-cli 12.55.2 v
+revision            0
+categories          devel
+platforms           darwin
+license             Apache-2
+maintainers         { ether.org.za:light @dylanbr } openmaintainer
+
+description         The official balena Command Line Interface.
+long_description    The balena CLI is a Command Line Interface for  \
+                    balenaCloud or openBalena. It builds on the balena API and \
+                    the balena SDK, and can also be directly imported in \
+                    Node.js applications.
+
+checksums           rmd160  abe76b9eb0b9ac62e07abce6c7b9d794df733386 \
+                    sha256  acdfdfca65e01bf036a93d9d042f25968413ffbb40113d39da97533867824c70 \
+                    size    708732
+
+depends_lib         path:bin/node:nodejs12
+depends_build       path:bin/npm:npm6
+
+configure.cmd       npm install
+configure.pre_args
+build.type          xcode
```

Comment:
Something else I've seen help with `node` builds is `use_xcode yes`.

Reply:
Thanks, I'll have to see if that help.

I need older versions of nodejs and npm for this and what seemed to be happening was that nodej16 was still being built because npm6 depends on it -- this doesn't happen locally if I install nodejs12 (or 14) first and then npm6.

Is there a way to prevent nodejs16 being built?

Reply:
Relevant part of the log

```
  Installing dependency (24 of 26) 'nodejs12' with variants '' (requesting '') ... [OK]
  Installing dependency (25 of 26) 'nodejs16' with variants '' (requesting '') ... [OK]
  Installing dependency (26 of 26) 'npm6' with variants '' (requesting '') ... [OK]
Error: Failed to install dependencies for balena-cli
Error: Process completed with exit code 1.
```

Reply:
On my machine in comparison

```
$ sudo port install nodejs12 npm6
--->  Computing dependencies for nodejs12
--->  Fetching archive for nodejs12
--->  Attempting to fetch nodejs12-12.22.7_2.darwin_20.x86_64.tbz2 from https://packages.macports.org/nodejs12
--->  Attempting to fetch nodejs12-12.22.7_2.darwin_20.x86_64.tbz2.rmd160 from https://packages.macports.org/nodejs12
--->  Installing nodejs12 @12.22.7_2
--->  Activating nodejs12 @12.22.7_2
--->  Cleaning nodejs12
--->  Computing dependencies for npm6
--->  Fetching archive for npm6
--->  Attempting to fetch npm6-6.14.15_0.darwin_20.noarch.tbz2 from https://packages.macports.org/npm6
--->  Attempting to fetch npm6-6.14.15_0.darwin_20.noarch.tbz2.rmd160 from https://packages.macports.org/npm6
--->  Installing npm6 @6.14.15_0
--->  Activating npm6 @6.14.15_0
--->  Cleaning npm6
--->  Scanning binaries for linking errors
--->  No broken files found.
--->  No broken ports found.
--->  Some of the ports you installed have notes:
  npm6 has the following notes:
    It is not recommended to install packages globally. But if you do so please be aware that they won't get cleaned up when you
    deactivate or uninstall npm6. Globally installed packages will remain in /opt/local/lib/node_modules/ until you manually delete
    them.
```

Reply:
Unfortunately that is going to be unavoidable:

```
$ port info npm6
npm6 @6.14.15 (devel)
...
Library Dependencies: nodejs16
...
```

`nodejs16` is specifically listed as being a dependency for `npm6`. Is it absolutely impossible to build this with much more recent `nodejs` and `npm` versions?

Reply:
Unfortunately not. It works with `nodejs14` (with some warnings) but won't build at all with `nodejs16`.